gstreamer-source.c

```diff
@@ -67,11 +68,50 @@ static gboolean start_pipe(gpointer user_data)
 	return G_SOURCE_REMOVE;
 }
 
+static void update_media_state(GstMessage *message, data_t *data)
+{
+	switch (GST_MESSAGE_TYPE(message)) {
+	case GST_MESSAGE_STATE_CHANGED:
+		GstState newstate;
+		gst_message_parse_state_changed(message,
+						NULL, &newstate, NULL);
+		switch (newstate) {
+		default: // Nb. GST_STATE_VOID_PENDING is not possible
+		case GST_STATE_NULL:
+			data->media_state = OBS_MEDIA_STATE_NONE;
+			break;
+		case GST_STATE_READY:
+			data->media_state = OBS_MEDIA_STATE_STOPPED;
+			break;
+		case GST_STATE_PAUSED:
+			data->media_state = OBS_MEDIA_STATE_PAUSED;
+			break;
+		case GST_STATE_PLAYING:
+			data->media_state = OBS_MEDIA_STATE_PLAYING;
+			break;
+		}
+		break;
+	case GST_MESSAGE_BUFFERING:
```

Comment:
I think it is required to parse the actual message. If the value from the message is `!= 100` it is basically buffering, when it is `== 100` when buffering is complete and flag should be cleared.

Reply:
Yes, I'll take another look at this.